core/services/keeper/validate.go

```diff
@@ -33,6 +33,18 @@ decode_check_upkeep_tx   [type=ethabidecode
 encode_perform_upkeep_tx [type=ethabiencode
                           abi="performUpkeep(uint256 id, bytes calldata performData)"
                           data="{\"id\": $(jobSpec.upkeepID),\"performData\":$(decode_check_upkeep_tx.performData)}"]
+simulate_perform_upkeep_tx  [type=ethcall
+                          extractRevertReason=true
+                          evmChainID="$(jobSpec.evmChainID)"
+                          contract="$(jobSpec.contractAddress)"
+                          from="$(jobSpec.fromAddress)"
+                          gas="$(jobSpec.performUpkeepGasLimit)"
+                          data="$(encode_perform_upkeep_tx)"]
```

Comment:
Just thought about this, should we also add 
```
                    gasPrice="$(jobSpec.gasPrice)"
                    gasTipCap="$(jobSpec.gasTipCap)"
                    gasFeeCap="$(jobSpec.gasFeeCap)"
```
to simulate the gas price as we do in checkUpkeep

Reply:
@RyanRHall i forgot why we left this out

Reply:
I would vote to leave this out of the simulation. I think adding gas prices to these eth_calls was a hacky feature-add and has been buggy and cumbersome. I'd also like to remove this "back door" feature entirely in the next release of keepers and instead pas gas price explicitly as an argument to checkUpkeep. As such, I'd prefer we don't get customers accustomed to using it.